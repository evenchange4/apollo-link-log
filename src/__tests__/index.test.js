// @flow
// $FlowFixMe
import gql from 'graphql-tag';
import { ApolloLink, execute, Observable, makePromise } from 'apollo-link';
import createLogLink from '../index';

it('default', async () => {
  const query = gql`
    query queryFoo {
      foo {
        bar
      }
    }
  `;

  const mockLink = new ApolloLink(() => Observable.of({ results: 'data' }));
  const link = (createLogLink: any)().concat(mockLink);
  try {
    const data = await makePromise(execute(link, { query }));
    expect(data).toEqual({ results: 'data' });
  } catch (error) {
    expect(error).toBeUndefined();
  }
});

it('custom logger', async () => {
  const query = gql`
    query queryFoo {
      foo {
        bar
      }
    }
  `;

  const mockLogger = jest.fn();
  const mockLink = new ApolloLink(() => Observable.of({ results: 'data' }));
  const link = createLogLink({
    logger: mockLogger,
  }).concat(mockLink);
  try {
    const data = await makePromise(execute(link, { query }));
    expect(data).toEqual({ results: 'data' });
    expect(mockLogger).toMatchSnapshot();
  } catch (error) {
    expect(error).toBeUndefined();
  }
});

it('disabled', async () => {
  const query = gql`
    query queryFoo {
      foo {
        bar
      }
    }
  `;

  const mockLogger = jest.fn();
  const mockLink = new ApolloLink(() => Observable.of({ results: 'data' }));
  const link = createLogLink({
    logger: mockLogger,
    enabled: false,
  }).concat(mockLink);
  try {
    const data = await makePromise(execute(link, { query }));
    expect(data).toEqual({ results: 'data' });
    expect(mockLogger).not.toHaveBeenCalled();
  } catch (error) {
    expect(error).toBeUndefined();
  }
});
