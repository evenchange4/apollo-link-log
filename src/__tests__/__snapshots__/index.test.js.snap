// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`custom logger 1`] = `
[MockFunction] {
  "calls": Array [
    Array [
      Object {
        "operation": Object {
          "extensions": Object {},
          "operationName": "queryFoo",
          "query": Object {
            "definitions": Array [
              Object {
                "directives": Array [],
                "kind": "OperationDefinition",
                "name": Object {
                  "kind": "Name",
                  "value": "queryFoo",
                },
                "operation": "query",
                "selectionSet": Object {
                  "kind": "SelectionSet",
                  "selections": Array [
                    Object {
                      "alias": undefined,
                      "arguments": Array [],
                      "directives": Array [],
                      "kind": "Field",
                      "name": Object {
                        "kind": "Name",
                        "value": "foo",
                      },
                      "selectionSet": Object {
                        "kind": "SelectionSet",
                        "selections": Array [
                          Object {
                            "alias": undefined,
                            "arguments": Array [],
                            "directives": Array [],
                            "kind": "Field",
                            "name": Object {
                              "kind": "Name",
                              "value": "bar",
                            },
                            "selectionSet": undefined,
                          },
                        ],
                      },
                    },
                  ],
                },
                "variableDefinitions": Array [],
              },
            ],
            "kind": "Document",
            "loc": Object {
              "end": 62,
              "start": 0,
            },
          },
          "variables": Object {},
        },
        "operationName": "queryFoo",
        "query": "query queryFoo {
  foo {
    bar
  }
}
",
        "variables": Object {},
      },
    ],
  ],
  "results": Array [
    Object {
      "isThrow": false,
      "value": undefined,
    },
  ],
}
`;
